numpy/linalg/_linalg.pyi

```diff
@@ -480,3 +487,27 @@ def matmul(
     x1: _ArrayLikeComplex_co,
     x2: _ArrayLikeComplex_co,
 ) -> NDArray[complexfloating]: ...
+
+@overload
+def geneigvals(a: _ArrayLikeInt_co, b: _ArrayLikeInt_co) -> NDArray[float64]: ...
+@overload
+def geneigvals(a: _ArrayLikeFloat_co, b: _ArrayLikeFloat_co) -> NDArray[floating]: ...
+@overload
+def geneigvals(a: _ArrayLikeComplex_co, b: _ArrayLikeComplex_co) -> NDArray[complexfloating]: ...
+
+@overload
+def geneig(
+    a: _ArrayLikeInt_co,
+    b: _ArrayLikeInt_co,
+) -> PolyEigResult: ...
+@overload
+def geneig(
+    a: _ArrayLikeFloat_co,
+    b: _ArrayLikeFloat_co,
+) -> PolyEigResult: ...
+
+@overload
```

Comment:
This is not an overloaded function

```suggestion
```

Overloading it is also an option, just like how you've done with `geneigvals`.